src/services/completions.ts

```diff
@@ -1866,6 +1884,7 @@ namespace ts.Completions {
                 return contextualMemberSymbols;
             }
 
+            const membersDeclaredBySpreadAssignment: Symbol[] = [];

```

Comment:
Instead of this make it `let membersDeclaredBySpreadAssignment: Symbol[] | undefined= [];` and then push as 
`(membersDeclaredBySpreadAssignment || (membersDeclaredBySpreadAssignment = [])).push(...properties);` and later pass as readonly array by `setSortTextToMemberDeclaredBySpreadAssignment(membersDeclaredBySpreadAssignment || emptyArray, filteredSymbols);`
This helps in common case of not having to allocate array if there is no spread assignment.